src/main/java/io/stargate/sgv2/jsonapi/service/operation/tasks/TaskBuilder.java

```diff
@@ -0,0 +1,75 @@
+package io.stargate.sgv2.jsonapi.service.operation.tasks;
+
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.DefaultDriverExceptionHandler;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.SchemaObject;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Extensible base for builders to create {@link Task} objects. */
+public abstract class TaskBuilder<TaskT extends Task<SchemaT>, SchemaT extends SchemaObject> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TaskBuilder.class);
+
+  // first value is zero, but we increment before we use it
+  private int taskPosition = -1;
+
+  // No default to make sure that we remember that set it to something specific if needed.
+  private DefaultDriverExceptionHandler.Factory<SchemaT> exceptionHandlerFactory = null;
+
+  protected final SchemaT schemaObject;
+
+  /** */
+  protected TaskBuilder(SchemaT schemaObject) {
+    this.schemaObject = Objects.requireNonNull(schemaObject, "schemaObject must not be null");
+  }
+
+  protected int nextPosition() {
+    return taskPosition += 1;
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T extends TaskBuilder<TaskT, SchemaT>> T withExceptionHandlerFactory(
```

Comment:
Similar to Accumulator, these are ugly casts: would it be possible to define SELF type (parameter) for builder case class instead?

Reply:
yes, I got the fluent pattern wrong all over the place :(  Started improving it in the next PR